File: z2python/zSchronisko.py
class Volunteer:
    def __init__(self, name: str, phone: str):
        self.name = name
        self.phone = phone

class Animal: 
    def __init__(self, name: str, species: str):
        self.name = name
        self.species = species
        self.caretaker = None
        self.is_adopted = False 

class Adoption:
    def __init__(self):
        self.animals = []

    def register_animal(self, animal):
        self.animals.append(animal)
        print(f"Zarejestrowano: {animal.name} (Gatunek: {animal.species})")

    def assign_caretaker(self, animal, volunteer):
        animal.caretaker = volunteer
        print(f"Wolontariusz {volunteer.name} (tel. {volunteer.phone}) został opiekunem zwierzaka o imieniu {animal.name}.")

    def process_adoption(self, animal):
        if not animal.is_adopted:
            animal.is_adopted = True
            print(f"SUKCES! Zwierzę {animal.name} znalazło nowy dom!")
        else:
            print(f"Zwierzę {animal.name} zostało już wcześniej zaadoptowane.")

# Testy
if __name__ == "__main__":
    vol1 = Volunteer("Jan Nowak", "123-456-789")
    vol2 = Volunteer("Adam Kowalski", "987-654-321")

    dog = Animal("Reksio", "Pies")
    cat = Animal("Kot w butach", "Kot")

    shelter = Adoption()

    print("REJESTRACJA")
    shelter.register_animal(dog)
    shelter.register_animal(cat)

    print("\nPRZYDZIAŁ OPIEKUNA")
    shelter.assign_caretaker(dog, vol1)
    shelter.assign_caretaker(cat, vol2)

    print("\nPROCES ADOPCJI")
    shelter.process_adoption(dog)

    shelter.process_adoption(dog) #Test ponownej adopcji 

File: z2C#/zSchronisko.cs
using System;

public class Volunteer
{
    public string Name { get; }
    public string Phone {  get; }

    public Volunteer(string name, string phone)
    {
        Name = name;
        Phone = phone;
    }
}

public class Animal
{
    public string Name { get; }
    public string Species { get; }
    public bool IsAdopted { get; set; }
    public Volunteer Caretaker { get; set; }

    public Animal(string name, string species)
    {
        Name = name;
        Species = species;
        IsAdopted = false;
        Caretaker = null;
    }
}

public class Adoption
{
    private readonly List<Animal> animals = new List<Animal>();

    public void RegisterAnimal(Animal animal)
    {
        animals.Add(animal);
        Console.WriteLine($"Zarejestrowano: {animal.Name} (Gatunek: {animal.Species})");
    }

    public void AssignCaretaker(Animal animal, Volunteer volunteer)
    {
        animal.Caretaker = volunteer;
        Console.WriteLine($"Wolontariusz {volunteer.Name} (tel. {volunteer.Phone}) został opiekunem zwierzaka o imieniu {animal.Name}");
    }

    public void ProcessAdoption(Animal animal)
    { 
        if (!animal.IsAdopted)
        {
            animal.IsAdopted = true;
            Console.WriteLine($"SUKCES! Zwierzę {animal.Name} znalazło nowy dom!");
        }
        else
        {
            Console.WriteLine($"Zwierzę {animal.Name} zostało już wcześniej zaadoptowane");
        }
    }
}
//Testy
class Program
{
    static void Main()
    {
        var vol1 = new Volunteer("Jan Nowak", "123-456-789");
        var vol2 = new Volunteer("Adam Kowalski", "987-654-321");

        var dog = new Animal("Reksio", "Pies");
        var cat = new Animal("Kot w butach", "Kot");

        var shelter = new Adoption();

        Console.WriteLine("REJESTRACJA");
        shelter.RegisterAnimal(dog);
        shelter.RegisterAnimal(cat);

        Console.WriteLine("\nPRZYDZIAŁ OPIEKUNA");
        shelter.AssignCaretaker(dog, vol1);
        shelter.AssignCaretaker(cat, vol2);

        Console.WriteLine("\nPROCES ADOPCJI");
        shelter.ProcessAdoption(dog);

        shelter.ProcessAdoption(dog); //Test ponownej adopcji
    }
}